### ISSUE TITLE
[bug] [feature request] No results when database is huge - introduce results paging

### ISSUE THREAD

**Issue by mgeeky** [dirkjanm/ROADtools#33]
Hi there Dirk!

Thank you for a splendid tool! :-)

I've just collected a rather big database (2,5GB) containing more than 100k users, 500k groups, 260k devices etc. 
The `gather` phase went just fine, but when I attempted to review results from GUI - no records could be rendered.

Whenever frontend app launches Ajax request, the server is unable to serialize JSON in a timely manner, making frontend wait extremely long for the response. This results in GUI displaying no records until server responds.

In my situation, the server was unable to come up with a response in 30 minutes.

Maybe there could be a paging functionality added with frontend API calling out to specific results-pages as a workaround?

Leaving this as a food for thought! :-)

Regards,
Mariusz.

**Comment by dirkjanm** [comment]
Hey,

I've run into this as well on large tenants. The problem here is that for all tenants except those extremely large ones it works fine to just load all the data and use in-memory filtering. 
If doing paging on the backend that would need to include searching and filtering as well. Which essentially means I have to rebuild a large part of the frontend just to accomodate those larger tenants.
This may be something I will do at some point, but at the moment it's not a high priority.

**Comment by mgeeky** [comment]
Yes, I agree - these changes would entail substantial upgrades to both frontend and backend. Yet I believe they would offer worth ROI while working in larger tenants. Currently I'm pretty much unable to efficiently explore data collected - and I'm yet to work with more of that large tenants in upcoming engagements. 

Anyway, I'm keeping my fingers crossed for you to find time to address this one! :-)

Regards,
M.

**Comment by rgmz** [comment]
> If doing paging on the backend that would need to include searching and filtering as well. Which essentially means I have to rebuild a large part of the frontend and backend just to accomodate those larger tenants.
> This may be something I will do at some point, but at the moment it's not a high priority.

@dirkjanm I've created a POC in #94 (only pagination for `/api/users`, no sorting). Have a look and let me know your thoughts.

**Comment by rgmz** [comment]
Hey @dirkjanm, let me know if there's interest in fleshing out my proposed fix. If not, I will close the PR.

**Comment by dirkjanm** [comment]
I've implemented this in the latest version of the UI, which now supports paging and builds on the previous work of @rgmz . thanks for the contributions and apologies I didn't merge this in earlier
